components/search/src/filter.rs

```diff
@@ -94,6 +95,17 @@ impl SearchEngineDefinition {
         }
     }
 
+    fn merge_override(&mut self, override_record: &JSONOverridesRecord) {
```

Comment:
It'd be nice to add a unit test for this - something fairly simple that checks we merge the parameters correctly (and don't change the telemetry suffix if none is supplied).